umbraco/Umbraco-CMS
# V8: Handle element types in the content editor

### Prerequisites


If there's an existing issue for this PR then this fixes: https://github.com/umbraco/Umbraco-CMS/issues/4168

### Description

Element types should not be created in the content tree. But when an existing content type is changed into an element type, chances are that element types *will* be present in the content tree. 

Thus the content editor needs to handle these "left over" element types gracefully.

This PR ensures that for element types in the content tree:

- The "Preview" and "Save and publish" buttons are removed.
- The URL list is removed.
- The template editing is removed.

![image](https://user-images.githubusercontent.com/7405322/51558363-8f938380-1e7f-11e9-9f2e-3480d0eeafe2.png)

![image](https://user-images.githubusercontent.com/7405322/51558486-d7b2a600-1e7f-11e9-86e0-1981aefbaa02.png)

#### To test this PR

1. Create some content .
2. Verify that the limitations mentioned above are not imposed on the content.
3. Change the content type into an element type.
4. Verify that the limitations mentioned above are imposed on the element.

#### Merge conflict warning!

This PR will most likely conflict with #4136, as they both impose limitations on some of the same things.
